lintro/ai/review/progress.py

```
@@ -198,3 +241,21 @@ def _passes_per_chunk(depth: int) -> int:
     return 1  # review only
+
+
+class StepTrackingProgress:
+    """Wraps a progress tracker and records the most recent step name."""
+
+    def __init__(self, inner: ReviewProgressCallback) -> None:
+        """Wrap ``inner`` and track the latest step name."""
+        self._inner = inner
+        self.last_step = "reviewing"
+
+    def __getattr__(self, name: str):  # noqa: ANN001
```

Ruff (0.15.20)

[warning] Missing return type annotation for special method `__getattr__`

(ANN204)